python/ray/air/session.py

```diff
@@ -1,14 +1,43 @@
-from typing import TYPE_CHECKING, Dict, Optional, Union
+from typing import TYPE_CHECKING, Any, Callable, Dict, Optional, Union
+import warnings
+import functools
 
 from ray.air._internal.session import _get_session
 from ray.air.checkpoint import Checkpoint
+from ray.air.constants import SESSION_MISUSE_LOG_ONCE_KEY
 from ray.train.session import _TrainSessionImpl
+from ray.util import log_once
 
 if TYPE_CHECKING:
     from ray.data import Dataset, DatasetPipeline
     from ray.tune.execution.placement_groups import PlacementGroupFactory
 
 
+def _warn_session_misuse(default_value: Any = None):
+    """Warns if fn is being used outside of session and returns ``default_value``."""
+
+    def inner(fn: Callable):
+        fn_name = fn.__name__
+
+        @functools.wraps(fn)
+        def wrapper(*args, **kwargs):
+            session = _get_session()
+            if not session:
+                if log_once(f"{SESSION_MISUSE_LOG_ONCE_KEY}-{fn_name}"):
+                    warnings.warn(
+                        f"`{fn_name}` is meant to only be "
+                        "called inside a function that is executed by a Tuner"
+                        " or Trainer."
```

Comment:
Can we explicitly say what we are returning? Keep this from the old warning message: `Returning None.`